posthog/api/hog_function.py

```diff
@@ -161,6 +161,9 @@ def to_internal_value(self, data):
         data["inputs_schema"] = data.get("inputs_schema", instance.inputs_schema if instance else [])
         data["inputs"] = data.get("inputs", instance.inputs if instance else {})
 
+        # Always ensure filters is initialized as an empty object if it's null
+        data["filters"] = data.get("filters") or {}
```

Comment:
check the code above - you need to pull from the instance if not set otherwise you are resetting this on all PATCH requests that don't include the value